docs/docs/authentication-tutorial.md

```diff
@@ -2,7 +2,445 @@
+
+```javascript:title=gatsby-config.js
+module.exports = {
+  siteMetadata: {
+    title: "Gatsby Authentication Tutorial",
+  },
+  plugins: ["gatsby-plugin-react-helmet", "gatsby-plugin-offline"],
+}
+```
+
+Create a new component to hold the links, and edit the layout component to include it. For now, it will act as a placeholder:
+
+```jsx:title=src/components/navBar.js
+import React from "react"
+import { Link } from "gatsby"
+
+export default () => (
+  <div
+    style={{
+      display: "flex",
+      flex: "1",
+      justifyContent: "space-between",
+      borderBottom: "1px solid #d1c1e0",
+    }}
+  >
+    <span>You are not logged in</span>
+
+    <nav>
+      <Link to="/">Home</Link>
+      {` `}
+      <Link to="/">Profile</Link>
+      {` `}
+      <Link to="/">Logout</Link>
+    </nav>
+  </div>
+)
+```
+
+```jsx{7,41}:title=src/components/layout.js
+import React from "react"
+import PropTypes from "prop-types"
+import Helmet from "react-helmet"
+import { StaticQuery, graphql } from "gatsby"
+
+import Header from "./header"
+import NavBar from "./navBar"
+import "./layout.css"
+
+const Layout = ({ children }) => (
+  <StaticQuery
+    query={graphql`
+      query SiteTitleQuery {
+        site {
+          siteMetadata {
+            title
+          }
+        }
+      }
+    `}
+    render={data => (
+      <>
+        <Helmet
+          title={data.site.siteMetadata.title}
+          meta={[
+            { name: "description", content: "Sample" },
+            { name: "keywords", content: "sample, something" },
+          ]}
+        >
+          <html lang="en" />
+        </Helmet>
+        <Header siteTitle={data.site.siteMetadata.title} />
+        <div
+          style={{
+            margin: "0 auto",
+            maxWidth: 960,
+            padding: "0px 1.0875rem 1.45rem",
+            paddingTop: 0,
+          }}
+        >
+          <NavBar />
+          {children}
+        </div>
+      </>
+    )}
+  />
+)
+
+Layout.propTypes = {
+  children: PropTypes.node.isRequired,
+}
+
+export default Layout
+```
+
+Lastly, change the index page to include this new content:
+
+```jsx{9-11}:title=src/pages/index.js
+import React from "react"
+import { Link } from "gatsby"
+
+import Layout from "../components/layout"
+
+const IndexPage = () => (
+  <Layout>
+    <h1>Hi people</h1>
+    <p>
+      You should <Link to="/">log in</Link> to see restricted content
+    </p>
+  </Layout>
+)
+
+export default IndexPage
+```
+
+## Authentication service
+
+For this tutorial you will use a hardcoded user/password. Create the folder `src/services` and add the following content to the file `auth.js`:
+
+```javascript:title=src/services/auth.js
+export const isBrowser = () => typeof window !== "undefined"
+
+export const getUser = () =>
+  isBrowser() && window.localStorage.gatsbyUser
+    ? JSON.parse(window.localStorage.gatsbyUser)
+    : {}
+
+const setUser = user => (window.localStorage.gatsbyUser = JSON.stringify(user))
```

Comment:
I honestly didn't even know this was possible. Maybe use the `getItem` and `setItem` API instead?

Reply:
Agreed. Better use the documented API to avoid compatibility issues later.